src/clj/puppetlabs/services/request_handler/request_handler_core.clj

```diff
@@ -259,73 +251,13 @@
   "Make the request mutable.  This is required by the ruby layer."
   (HashMap. request))
 
-(defn bad-request?
-  [x]
-  "Determine if the supplied slingshot message is for a 'bad request'"
-  (when (map? x)
-    (= (:type x)
-       :puppetlabs.services.request-handler.request-handler-core/bad-request)))
-
-(defn service-unavailable?
-  [x]
-  "Determine if the supplied slingshot message is for a 'service unavailable'"
-  (when (map? x)
-    (= (:type x)
-       :puppetlabs.services.jruby.jruby-puppet-service/service-unavailable)))
-
-(defn jruby-timeout?
-  "Determine if the supplied slingshot message is for a JRuby borrow timeout."
-  [x]
-  (when (map? x)
-    (= (:type x)
-       :puppetlabs.services.jruby.jruby-puppet-service/jruby-timeout)))
-
-(defn output-error
-  [{:keys [uri]} {:keys [message]} http-status]
-  (log/errorf "Error %d on SERVER at %s: %s" http-status uri message)
-  (-> (ring-response/response message)
-      (ring-response/status http-status)
-      (ring-response/content-type "text/plain")))
-
-(defn wrap-with-error-handling
-  "Middleware that wraps a JRuby request with some error handling to return
-  the appropriate http status codes, etc."
-  [f]
-  (fn [request]
-    (sling/try+
-      (f request)
-      (catch bad-request? e
-        (output-error request e 400))
-      (catch jruby-timeout? e
-        (output-error request e 503))
-      (catch service-unavailable? e
-        (output-error request e 503)))))
-
-(defn wrap-with-jruby-instance
-  "Middleware fn that borrows a jruby instance from the `jruby-service` and makes
-  it available in the request as `:jruby-instance`"
-  [f jruby-service]
-  (fn [request]
-    (jruby/with-jruby-puppet
-      jruby-instance
-      jruby-service
-      {:request (dissoc request :ssl-client-cert)}
-      
-      (f (assoc request :jruby-instance jruby-instance)))))
```

Comment:
Yeah, I didn't change anything here from before - just moved to a new namespace.  Which part did you think was scary?  The idea of adding a `jruby-instance` onto the Ring request map for the next function in the middleware chain to use - or were you talking about the line that removes the `:ssl-client-cert` from the Ring request map, presumably so that isn't surfaced up to the metrics / status APIs?